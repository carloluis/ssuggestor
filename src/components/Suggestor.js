import React, { Component } from 'react';
import withClickOut from './withClickOut';

const SPIN_STYLES = {
	position: 'absolute',
	cursor: 'pointer',
	margin: 'auto',
	color: '#ccc',
	fontSize:16,
	right:10,
	top: 0,
	bottom: 0,
	height: 14,
	zIndex:4
};
const X_STYLES = {
	...SPIN_STYLES,
	fontSize: 14,
	right: 30
};

const getListStyles = visible => ({
	display: visible? 'block':'none',
	maxHeight: 190,
	minWidth: 30,
	width: 'inherit',
	overflow: 'auto',
	cursor: 'pointer'
});

const KEY_CODES = {
	ENTER: 13,
	ESCAPE: 27,
	DOWN: 40,
	UP: 38
};

const EMPTY_STR = '';

const ListItem = ({ item, onItemClick, onItemMouseEnter, index, overItem }) => (
	<li value={item} onClick={() => onItemClick(item)} onMouseEnter={()=>onItemMouseEnter(index, item)} 
		style={{ backgroundColor: overItem && '#f5f5f5' }}>
		<a>{item}</a>
	</li>
);

const List = ({ open, list, index, onItemClick, onItemMouseEnter }) => (
	<ul className="dropdown-menu" style={getListStyles(open)}>
		{ list.map((item, i) => <ListItem key={i} {...{ item, onItemClick, onItemMouseEnter, index:i, overItem:index===i }} />) }
	</ul>
);

export class Suggestor extends Component {
	constructor(props){
		super(props);
		this.state = {
			value: EMPTY_STR,
			open: false,
			index: 0
		};
		this.handleClickOut = this.handleClickOut.bind(this);
		this.handleClick = this.handleClick.bind(this);
		this.handleChange = this.handleChange.bind(this);
		this.removeItem = this.removeItem.bind(this);
		this.handleKeyDown = this.handleKeyDown.bind(this);
		this.handleItemClick = this.handleItemClick.bind(this);
		this.handleItemMouseEnter = this.handleItemMouseEnter.bind(this);
	}
	handleClickOut(){
		this.setState({ open: false });
	}
	handleClick(e){
		this.setState({ open: !this.state.open });
	}
	handleKeyDown(e){
		let { open, index, value } = this.state;
		let list = this.filter(this.props.list, value);

		switch (e.keyCode) {
			case KEY_CODES.ENTER:
				this.setState({ open: !open, value: list[index] });
				break;
			case KEY_CODES.ESCAPE:
				this.setState({ open: false, value: open? value: EMPTY_STR });
				break;
			case KEY_CODES.DOWN:
				let next = (index + open) % list.length;
				this.setState({ open: true, index: next });
				break;
			case KEY_CODES.UP:
				let prev = (index || list.length) - 1;
				this.setState({ open: true, index: prev });
				break;
			default:
				return;
		}

		e.preventDefault();
	}
	handleItemClick(value){
		this.setState({ value });
	}
	handleItemMouseEnter(index){
		this.setState({ index });
	}
	handleChange(e){
		e.stopPropagation();
		this.setState({ value: e.target.value, open: true });
	}
	removeItem(){
		this.setState({ value: EMPTY_STR });
	}
	filter() {
		return this.props.list.filter(item => item.includes(this.state.value));
	}
	render() {
		let { open, value, index } = this.state;
		let { style, placeholder } = this.props;
		let list = this.filter();

		return (
			<div className="input-group" style={style} onClick={this.handleClick} onKeyDown={this.handleKeyDown} tabIndex="0" >
				<input type="text" className="form-control" onChange={this.handleChange} value={value} placeholder={placeholder} />
				<span className="glyphicon glyphicon-triangle-bottom" style={SPIN_STYLES} />
				{ !open && value && <span className="glyphicon glyphicon-remove" style={X_STYLES} onClick={this.removeItem}/> }
				<List list={list} open={open} index={index} 
					onItemClick={this.handleItemClick} 
					onItemMouseEnter={this.handleItemMouseEnter} />
			</div>
		);
	}
}

Suggestor.propTypes = {
	list: React.PropTypes.arrayOf(React.PropTypes.string).isRequired,
	item: React.PropTypes.string,
	style: React.PropTypes.object
};
Suggestor.defaultProps = {
	item: EMPTY_STR
};

export default withClickOut(Suggestor);