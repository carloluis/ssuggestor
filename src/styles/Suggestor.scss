@import './_palette.scss';

:global .ss-root {
	width: 100%;
}

:global .ss-input {
	padding-right: 42px;
}

:global .ss-arrow,
:global .ss-remove {
	position: absolute;
	cursor: pointer;
	color: $grey;
	right: 6px;
	top: 50%;
	transform: translateY(-50%);
	z-index: 4;
}

:global .ss-remove {
	right: 26px;
	&::after {
		content: '\D7';
		font-size: 16px;
		font-weight: 500;
	}
}

:global .ss-arrow {
	border-top: 8px solid $grey;
	border-right: 6px solid transparent;
	border-left: 6px solid transparent;
	border-bottom: 0;
	margin: 2px;
}
