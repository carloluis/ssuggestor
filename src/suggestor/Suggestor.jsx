import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { autoBind, keys, noop } from '../utils';
import transform from '../utils/transform';
import ListItem from './ListItem';

class Suggestor extends PureComponent {
	constructor(props) {
		super(props);
		autoBind(this);

		this.input = React.createRef();

		this.state = {
			filtered: this.filter(props.list, props.value, false),
			value: props.value,
			open: false,
			index: 0
		};
	}
	componentDidMount() {
		document.addEventListener('click', this._onClick);
	}
	componentWillUnmount() {
		document.removeEventListener('click', this._onClick);
	}
	_onClick(event) {
		if (!this.input.current.parentNode.contains(event.target)) {
			this.close();
		}
	}
	componentWillReceiveProps(nextProps) {
		let value = this.state.value;

		if (nextProps.value !== this.props.value && nextProps.value !== value) {
			value = nextProps.value;
		}

		this.setState({
			filtered: this.filter(nextProps.list, value, true),
			value
		});
	}
	close() {
		this.setState({
			open: false,
			filtered: this.unfilter(),
			index: 0
		});
	}
	handleClick() {
		if (this.props.openOnClick) {
			if (this.state.open) {
				this.close();
			} else {
				this.setState({ open: true, filtered: this.unfilter() });
			}
		}
	}
	handleKeyDown(e) {
		const { onKey, useKeys } = this.props;
		onKey(e);

		if (useKeys && this.processKey(e.keyCode)) {
			e.preventDefault();
		}
	}
	processKey(code) {
		const { open, index, filtered, value } = this.state;
		const ssuggestions = filtered.length ? filtered : this.unfilter();
		let nextIndex;

		switch (code) {
			case keys.ENTER:
				if (open && filtered[index]) {
					this.changeValue(filtered[index].word, true);
				} else {
					this.setState({ open: true, filtered: this.unfilter() });
				}
				break;
			case keys.ESCAPE:
				this.close();
				if (!open && value) {
					this.changeValue('');
				}
				break;
			case keys.DOWN:
				nextIndex = (index + open) % ssuggestions.length;
				break;
			case keys.UP:
				nextIndex = (index || ssuggestions.length) - 1;
				break;
			case keys.TAB:
				if (this.props.selectOnTab && open && filtered[index]) {
					this.changeValue(filtered[index].word, true);
				} else {
					this.close();
				}
			default:
				return false;
		}

		if (nextIndex !== undefined) {
			this.setState({ open: true, index: nextIndex, filtered: ssuggestions });
		}

		return true;
	}
	handleItemClick({ word }) {
		this.changeValue(word, true);
	}
	handleItemMouseEnter(index) {
		this.setState({ index });
	}
	handleChange(e) {
		e.stopPropagation();
		const value = e.target.value;
		this.changeValue(value);
	}
	remove() {
		this.changeValue('', true);
	}
	changeValue(value, select = false) {
		const { list, suggestOn, accents, onChange, onSelect } = this.props;
		const filtered = this.filter(list, value);
		const suggest = value.length >= suggestOn;
		const open = !!filtered.length && suggest;

		this.setState({ value, filtered, open }, () => {
			onChange(value);
			if (select) {
				const suggestion = filtered.find(({ word }) => transform(accents, word) === transform(accents, value));
				onSelect(value, suggestion && suggestion.item);
				this.close();
			}
		});
	}
	filter(list, value, onlyMatch = true) {
		const { accents, selector } = this.props;
		value = transform(accents, value);

		let mapped = list.map(item => {
			const word = selector(item);
			return {
				index: transform(accents, word).indexOf(value),
				word,
				item
			};
		});
		if (onlyMatch) {
			mapped = mapped.filter(item => item.index !== -1);
		}
		return mapped;
	}
	unfilter() {
		return this.filter(this.props.list, this.state.value, false);
	}
	focus() {
		this.input.current.focus();
	}
	render() {
		const { classSchema, style, placeholder, arrow, close, tooltip, required } = this.props;
		const { open, value, index, filtered } = this.state;
		const displaySuggestions = open && !!filtered.length;

		return (
			<div className={classSchema.root} onClick={this.handleClick} onKeyDown={this.handleKeyDown} style={style}>
				<input
					type="text"
					className={classSchema.input}
					onChange={this.handleChange}
					value={value}
					title={tooltip}
					placeholder={placeholder}
					required={required}
					ref={this.input}
				/>
				{arrow && <span className={classSchema.arrow} />}
				{close && value && <span className={classSchema.close} onClick={this.remove} />}
				{displaySuggestions && (
					<ul className={classSchema.list}>
						{filtered.map((item, i) => (
							<ListItem
								key={item.word}
								classSchema={classSchema}
								item={item}
								index={i}
								onItemClick={this.handleItemClick}
								onItemMouseEnter={this.handleItemMouseEnter}
								overItem={i === index}
								search={value}
							/>
						))}
					</ul>
				)}
			</div>
		);
	}
}

Suggestor.propTypes = {
	list: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string, PropTypes.object])).isRequired,
	selector: PropTypes.func,
	onChange: PropTypes.func,
	onSelect: PropTypes.func,
	onKey: PropTypes.func,
	value: PropTypes.string,
	openOnClick: PropTypes.bool,
	selectOnTab: PropTypes.bool,
	placeholder: PropTypes.string,
	tooltip: PropTypes.string,
	classSchema: PropTypes.shape({
		root: PropTypes.string,
		arrow: PropTypes.string,
		close: PropTypes.string,
		list: PropTypes.string,
		item: PropTypes.string,
		activeItem: PropTypes.string
	}),
	suggestOn: PropTypes.number,
	style: PropTypes.object,
	required: PropTypes.bool,
	useKeys: PropTypes.bool,
	accents: PropTypes.bool,
	arrow: PropTypes.bool,
	close: PropTypes.bool
};

Suggestor.defaultProps = {
	classSchema: {},
	selector: s => s,
	onSelect: noop,
	onChange: noop,
	onKey: noop,
	value: '',
	openOnClick: true,
	selectOnTab: false,
	suggestOn: 1,
	required: false,
	accents: false,
	useKeys: true,
	arrow: true,
	close: true
};

export default Suggestor;
