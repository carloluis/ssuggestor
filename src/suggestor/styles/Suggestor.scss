:global .ss-triangle,
:global .ss-remove {
	position: absolute;
	cursor: pointer;
	margin: auto;
	color: #ccc;
	font-size: 16px;
	right: 10px;
	top: 0;
	bottom: 0;
	height: 14px;
	z-index: 4;
}

:global .ss-remove {
	font-size: 14px;
	right: 30px;
}
