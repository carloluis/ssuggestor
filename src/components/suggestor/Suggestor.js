import React, { Component } from 'react';
import withClickOut from '../withClickOut';
import { SPIN_STYLES, X_STYLES } from './styles';
import List from './List';

const EMPTY_STR = '';

const KEY_CODES = {
	TAB: 9,
	ENTER: 13,
	ESCAPE: 27,
	DOWN: 40,
	UP: 38
};

export class Suggestor extends Component {
	constructor(props){
		super(props);
		this._bind('handleClick', 'handleChange', 'removeItem', 'handleKeyDown', 'handleItemClick', 'handleItemMouseEnter', 'handleClickOut', 'focus');

		this.state = {
			value: props.value,
			filtered: props.list,
			open: false,
			index: 0
		};
	}
	_bind(...methods) {
		methods.forEach(method => this[method] = this[method].bind(this));
	}
	handleClickOut() {
		this.handleClose();
	}
	componentWillReceiveProps(nextProps) {
		if (nextProps.value !== this.props.value) {
			this.setState({ value: nextProps.value });
		}
	}
	handleClose() {
		this.setState({ open: false, filtered: this.props.list, index: 0 });
	}
	toggleList() {
		if (this.state.open) {
			this.handleClose();
		}
		else {
			this.setState({
				open: true
			});
		}
	}
	handleClick() {
		if (this.props.openOnClick) {
			this.toggleList();
		}
	}
	handleKeyDown(e) {
		let { open, index, filtered:list } = this.state;

		switch (e.keyCode) {
			case KEY_CODES.TAB:
				if (this.props.selectOnTab && open && list[index]) {
					this.changeValue(list[index]);
				}
				this.handleClose();
				return;
			case KEY_CODES.ENTER:
				this.toggleList();
				if (open && list[index]) {
					this.changeValue(list[index]);	
				}
				break;
			case KEY_CODES.ESCAPE:
				this.handleClose();
				if (!open) {
					this.changeValue(EMPTY_STR);
				}
				break;
			case KEY_CODES.DOWN: {
				let next = (index + open) % list.length;
				this.setState({ open:true, index: next });
				break;
			}
			case KEY_CODES.UP: {
				let prev = (index || list.length) - 1;
				this.setState({ open: true, index: prev });
				break;
			}
			default:
				return;
		}

		e.preventDefault();
	}
	handleItemClick(value) {
		this.changeValue(value);
		if(!this.props.openOnClick) {
			this.toggleList();
		}
	}
	handleItemMouseEnter(index) {
		this.setState({ index });
	}
	handleChange(e) {
		e.stopPropagation();
		let value = e.target.value;
		let open = value.length >= this.props.suggestOn;
		let filtered = open? this.filter(value): this.state.filtered;
		this.setState({ open, filtered });
		if(!filtered.length) {
			this.handleClose();
		}
		this.changeValue(value);
	}
	removeItem() {
		this.changeValue(EMPTY_STR);
	}
	changeValue(value) {
		this.setState({ value });
		this.props.onChange(value);
	}
	filter(value) {
		value = value.toLowerCase();
		return this.props.list.filter(item => item.toLowerCase().indexOf(value) !== -1);
	}
	focus() {
		this.refs.input.focus();
	}
	render() {
		let { style, placeholder, arrow, nox, className } = this.props;
		let { open, value, index, filtered:list } = this.state;

		return (
			<div className={className} style={style} onClick={this.handleClick} onKeyDown={this.handleKeyDown} ref={this.props.reference}>
				<input type="text" className="form-control" onChange={this.handleChange} value={value} placeholder={placeholder} ref="input"/>
				{ arrow && <span className="glyphicon glyphicon-triangle-bottom" style={SPIN_STYLES} /> }
				{ !nox && value && <span className="glyphicon glyphicon-remove" style={X_STYLES} onClick={this.removeItem}/> }
				<List {...{ list, index, open, value }} onItemClick={this.handleItemClick} onItemMouseEnter={this.handleItemMouseEnter} />
			</div>
		);
	}
}

Suggestor.propTypes = {
	list: React.PropTypes.arrayOf(React.PropTypes.string).isRequired,
	reference: React.PropTypes.func,
	onChange: React.PropTypes.func,
	value: React.PropTypes.string,
	openOnClick: React.PropTypes.bool,
	selectOnTab: React.PropTypes.bool,
	suggestOn: React.PropTypes.number,
	placeholder: React.PropTypes.string,
	className: React.PropTypes.string,
	style: React.PropTypes.object,
	arrow: React.PropTypes.bool,
	nox: React.PropTypes.bool,
};
Suggestor.defaultProps = {
	className: 'input-group',
	onChange: f => f,
	value: EMPTY_STR,
	openOnClick: true,
	selectOnTab: false,
	suggestOn: 0,
	arrow: true,
	nox: false
};

export default withClickOut(Suggestor);