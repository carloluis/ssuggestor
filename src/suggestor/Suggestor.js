import React, { PureComponent } from 'react';
import { EMPTY_STR, KEY_CODES } from '../utils/values';
import { SPIN_STYLES, X_STYLES } from './styles';
import withClickOut from '../utils/withClickOut';
import List from './List';

export class Suggestor extends PureComponent {
	constructor(props){
		super(props);
		this._bind('handleClick', 'handleChange', 'remove', 'handleKeyDown', 'handleItemClick', 'handleItemMouseEnter', 'handleClickOut', 'focus');

		this.state = {
			value: props.value,
			filtered: props.list,
			open: false,
			index: 0
		};
	}
	_bind(...methods) {
		methods.forEach(method => this[method] = this[method].bind(this));
	}
	handleClickOut() {
		this.handleClose();
	}
	componentWillReceiveProps(nextProps) {
		if (nextProps.value !== this.state.value) {
			this.setState({ value: nextProps.value });
		}
	}
	handleClose() {
		this.setState({ open: false, filtered: this.props.list, index: 0 });
	}
	toggleList() {
		if (this.state.open) {
			this.handleClose();
		}
		else {
			this.setState({
				open: true
			});
		}
	}
	handleClick() {
		if (this.props.openOnClick) {
			this.toggleList();
		}
	}
	handleKeyDown(e) {
		this.props.onKey(e);
		if (!this.props.useKeys) {
			return;
		}
		let { open, index, filtered:list, value } = this.state;

		switch (e.keyCode) {
			case KEY_CODES.TAB:
				if (this.props.selectOnTab && open && list[index]) {
					this.changeValue(list[index], true);
				}
				this.handleClose();
				return;
			case KEY_CODES.ENTER:
				this.toggleList();
				if (open && list[index]) {
					this.changeValue(list[index], true);
				} else {
					this.props.onSelect(value);
				}
				break;
			case KEY_CODES.ESCAPE:
				this.handleClose();
				if (!open) {
					this.changeValue(EMPTY_STR);
				}
				break;
			case KEY_CODES.DOWN: {
				let next = (index + open) % list.length;
				this.setState({ open:true, index: next });
				break;
			}
			case KEY_CODES.UP: {
				let prev = (index || list.length) - 1;
				this.setState({ open: true, index: prev });
				break;
			}
			default:
				return;
		}

		e.preventDefault();
	}
	handleItemClick(value) {
		this.changeValue(value, true);
		if(!this.props.openOnClick) {
			this.toggleList();
		}
	}
	handleItemMouseEnter(index) {
		this.setState({ index });
	}
	handleChange(e) {
		e.stopPropagation();
		let value = e.target.value;
		let suggest = value.length >= this.props.suggestOn;
		let filtered = this.filter(value);
		let nextState = { filtered };
		if(!filtered.length || !suggest) {
			this.handleClose();
		} else{
			nextState.open = true;
		}
		this.setState(nextState);
		this.changeValue(value);
	}
	remove() {
		this.changeValue(EMPTY_STR, true);
	}
	changeValue(value, select=false) {
		this.setState({ value }, () => {
			this.props.onChange(value);
			if(select) {
				this.props.onSelect(value);
			}
		});
	}
	filter(value) {
		value = value.toLowerCase();
		return this.props.list.filter(item => item.toLowerCase().indexOf(value) !== -1);
	}
	focus() {
		this.refs.input.focus();
	}
	render() {
		let { className, style, placeholder, arrow, close, tooltip, required } = this.props;
		let { open, value, index, filtered:list } = this.state;

		return (
			<div className={className} style={style} onClick={this.handleClick} onKeyDown={this.handleKeyDown} ref={this.props.reference} >
				<input type="text" className="form-control" onChange={this.handleChange} value={value} ref="input" placeholder={placeholder} title={tooltip} required={required} />
				{ arrow && <span className="glyphicon glyphicon-triangle-bottom" style={SPIN_STYLES} /> }
				{ close && value && <span className="glyphicon glyphicon-remove" style={X_STYLES} onClick={this.remove}/> }
				<List {...{ list, index, open, value }} onItemClick={this.handleItemClick} onItemMouseEnter={this.handleItemMouseEnter} />
			</div>
		);
	}
}

Suggestor.propTypes = {
	list: React.PropTypes.arrayOf(React.PropTypes.string).isRequired,
	reference: React.PropTypes.func,
	onChange: React.PropTypes.func,
	onSelect: React.PropTypes.func,
	onKey: React.PropTypes.func,
	value: React.PropTypes.string,
	openOnClick: React.PropTypes.bool,
	selectOnTab: React.PropTypes.bool,
	placeholder: React.PropTypes.string,
	tooltip: React.PropTypes.string,
	className: React.PropTypes.string,
	suggestOn: React.PropTypes.number,
	style: React.PropTypes.object,
	required: React.PropTypes.bool,
	useKeys: React.PropTypes.bool,
	arrow: React.PropTypes.bool,
	close: React.PropTypes.bool
};

const nop = _ => _;

Suggestor.defaultProps = {
	className: 'input-group',
	onSelect: nop,
	onChange: nop,
	onKey: nop,
	value: EMPTY_STR,
	openOnClick: true,
	selectOnTab: false,
	suggestOn: 1,
	required: false,
	useKeys: true,
	arrow: true,
	close: true
};

export const SSuggestor = withClickOut(Suggestor);

export default SSuggestor;