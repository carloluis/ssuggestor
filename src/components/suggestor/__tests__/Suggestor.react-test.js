import React from 'react';
import renderer from 'react-test-renderer';
import Suggestor from '../Suggestor';
import ReactTestUtils from 'react-addons-test-utils';

describe('<Suggestor />', () => {
	it('initial render with no suggestions', () => {
		const tree = renderer.create(<Suggestor list={[]} />);
		expect(tree).toMatchSnapshot();
	});
	it('initial render with suggestions', () => {
		const tree = renderer.create(<Suggestor list={["suggest-1"]} />);
		expect(tree).toMatchSnapshot();
	});
	it('initial render with suggestions (open list)', () => {
		const tree = renderer.create(<Suggestor list={["suggest-1", "suggest-2"]} />);		
		tree.getInstance().refs.wrapped.handleClick();
		expect(tree).toMatchSnapshot();
	});
});

const LIST = ['suggest-1', 'suggest-2'];

test('Suggestor renders with closed list (open:false)', () => {
	// setup: arrange
	const component = ReactTestUtils.renderIntoDocument(<Suggestor list={LIST} />);

	// assert
	expect(component.refs.wrapped.state.open).toBeFalsy();
});

test('Suggestor change input\'s value => stores in state', () => {
	// setup: arrange
	const value = 'suggest';
	const component = ReactTestUtils.renderIntoDocument(<Suggestor list={LIST} />);

	// execute (act)
	let input = ReactTestUtils.findRenderedDOMComponentWithTag(component, 'input');
	input.value = value;
	ReactTestUtils.Simulate.change(input);

	// assert
	expect(component.refs.wrapped.state.value).toBe(value);
});

test('Suggestor change value => open suggestions list', () => {
	// setup: arrange & precondition...
	const component = ReactTestUtils.renderIntoDocument(<Suggestor list={LIST} />);

	// execute (act)
	let input = ReactTestUtils.findRenderedDOMComponentWithTag(component, 'input');
	input.value = 'suggest';
	ReactTestUtils.Simulate.change(input);

	// assert
	expect(component.refs.wrapped.state.open).toBeTruthy();
});
