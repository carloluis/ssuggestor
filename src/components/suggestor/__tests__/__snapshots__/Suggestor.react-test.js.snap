exports[`<Suggestor /> initial render with no suggestions 1`] = `
<div
  className="input-group"
  onClick={[Function]}
  onKeyDown={[Function]}
  style={undefined}>
  <input
    className="form-control"
    onChange={[Function]}
    placeholder={undefined}
    type="text"
    value="" />
  <span
    className="glyphicon glyphicon-triangle-bottom"
    style={
      Object {
        "bottom": 0,
        "color": "#ccc",
        "cursor": "pointer",
        "fontSize": 16,
        "height": 14,
        "margin": "auto",
        "position": "absolute",
        "right": 10,
        "top": 0,
        "zIndex": 4,
      }
    } />
  
</div>
`;

exports[`<Suggestor /> initial render with suggestions 1`] = `
<div
  className="input-group"
  onClick={[Function]}
  onKeyDown={[Function]}
  style={undefined}>
  <input
    className="form-control"
    onChange={[Function]}
    placeholder={undefined}
    type="text"
    value="" />
  <span
    className="glyphicon glyphicon-triangle-bottom"
    style={
      Object {
        "bottom": 0,
        "color": "#ccc",
        "cursor": "pointer",
        "fontSize": 16,
        "height": 14,
        "margin": "auto",
        "position": "absolute",
        "right": 10,
        "top": 0,
        "zIndex": 4,
      }
    } />
  
</div>
`;
